service/src/main/java/com/fecklessweasel/service/CourseServlet.java

```diff
@@ -59,4 +81,26 @@ public Integer run(Connection connection)
         // Redirect to homepage.
         response.sendRedirect("/course/index.jsp?cid=" + courseID);
     }
-}
+
+    @Override
+    protected void doGet(final HttpServletRequest request,
+                          final HttpServletResponse response)
+            throws ServletException, IOException {
+
+        final String username = request.getParameter("username");
+        final int courseID = OMUtil.parseInt(request.getParameter("cid"));
+
+        Boolean b =
+        SQLSource.interact(new SQLInteractionInterface<Boolean>() {
+            @Override 
+            public Boolean run(Connection connection)
+                throws ServiceException {
+  
+                final User user = User.lookup(connection, username);
+                return (Boolean) user.toggleFavoriteCourse(connection, courseID);
+            }
+        });
+
+        response.sendRedirect("/course/index.jsp?cid=" + courseID);
```

Comment:
If anyone has a clue on how to ajaxify this process, it'll probably look better and smoother than having to redirect back to the same page